core.d.ts

```diff
@@ -80,7 +80,7 @@ declare namespace PinejsClientCoreFactory {
 	}
 
 	export interface FilterArray extends Array<Filter> {}
-	export type FilterString = string
+	export type FilterBaseType = string | number | null | boolean | undefined | Date
```

Comment:
`undefined` should be left where it was before, it's only there for `FilterObj` to allow us to type the keys as optional, it's not a valid entry in a place where a filter is required (`null` is valid as it maps to a null in the api)